authors/apps/report/views.py

```diff
@@ -0,0 +1,136 @@
+from django.core.exceptions import ObjectDoesNotExist
+from django.shortcuts import render
+
+from rest_framework import status
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+from rest_framework.viewsets import ModelViewSet
+
+from authors.apps.articles.models import Article
+from authors.apps.profiles.models import Profile
+from authors.apps.report.models import ReportArticle
+from authors.apps.report.serializers import ReportArticleSerializer
+
+# Create your views here.
+class ReportArticleViewSet (ModelViewSet):
+    permission_classes = (IsAuthenticated,)
+    queryset = ReportArticle.objects.all().order_by('submission_date')
+    serializer_class = ReportArticleSerializer
+    
+    def list(self, request):
+        current_user = Profile.objects.get(user=request.user)
+        if not request.user.is_superuser:
+            self.queryset = ReportArticle.objects.all().filter(reporter=current_user)\
+            .order_by('submission_date')
+        serializer = ReportArticleSerializer(self.queryset, many=True)
+        return Response({"reports": serializer.data})
+
+    """
+    override retrieve method from the RetrieveModelMixin class
+    """
+    def retrieve(self, request, *args, **kwargs):
+        try:
+            reported_article = ReportArticle.objects.get(id=kwargs.get("report_id"))
+            serializer = self.get_serializer(reported_article)
+            current_user = Profile.objects.get(user=request.user)
+            if request.user.is_superuser or reported_article.reporter == current_user:
+                return Response ({"report":serializer.data})
+            else:
+                return Response (
+                    {"error":"You don't have the permissions to perform this request"},
+                    status=status.HTTP_401_UNAUTHORIZED
+                )
```

Comment:
Great work @llwasampijja, however, the `HTTP_401_UNAUTHORIZED` status code should be used explicitly for users that aren't logged in, in this case a user has access to the platform however, they aren't allowed to access this route, so the `HTTP_403_FORBIDDEN` status code best fits this permission check. Thank you.

Reply:
@AtamaZack ok, let me change that!